src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathMapperTest.java

```diff
@@ -786,6 +786,169 @@ public void testFourthCaseGroup() throws Exception {
+        final String xpath = "//METHOD_DEF/following-sibling::RCURLY";
+        final RootNode rootNode = getRootNode("InputXpathMapperAst.java");
+        final DetailAST[] actual = convertToArray(getXpathItems(xpath, rootNode));
+        final DetailAST expectedRightCurlyNode = getSiblingByType(rootNode.getUnderlyingNode(),
+                TokenTypes.CLASS_DEF)
+                .findFirstToken(TokenTypes.OBJBLOCK)
+                .findFirstToken(TokenTypes.METHOD_DEF)
+                .getNextSibling().getNextSibling();
+        final DetailAST[] expected = {expectedRightCurlyNode};
+        assertArrayEquals("Result nodes differ from expected", expected, actual);
+    }
+
+    @Test
+    public void testQueryElementFollowing() throws Exception {
+        final String xpath = "//IDENT[@text='variable']/following::*";
+        final RootNode rootNode = getRootNode("InputXpathMapperAst.java");
+        final DetailAST[] actual = convertToArray(getXpathItems(xpath, rootNode));
+        final DetailAST expectedAssignNode = getSiblingByType(rootNode.getUnderlyingNode(),
+                TokenTypes.CLASS_DEF)
+                .findFirstToken(TokenTypes.OBJBLOCK)
+                .findFirstToken(TokenTypes.METHOD_DEF)
+                .findFirstToken(TokenTypes.SLIST)
+                .findFirstToken(TokenTypes.VARIABLE_DEF)
+                .findFirstToken(TokenTypes.MODIFIERS)
+                .getNextSibling()
+                .getNextSibling()
+                .getNextSibling();
+        final DetailAST expectedExprNode = expectedAssignNode.getFirstChild();
+        final DetailAST expectedNumIntNode = expectedExprNode.getFirstChild();
+        final DetailAST[] expected = {expectedAssignNode, expectedExprNode, expectedNumIntNode};
+        assertArrayEquals("Result nodes differ from expected", expected, actual);
+    }
+
+    @Test
+    public void testQueryElementFollowingMethodDef() throws Exception {
+        final String xpath = "//PACKAGE_DEF/following::METHOD_DEF";
+        final RootNode rootNode = getRootNode("InputXpathMapperAst.java");
+        final DetailAST[] actual = convertToArray(getXpathItems(xpath, rootNode));
+        final DetailAST expectedMethodDefNode = getSiblingByType(rootNode.getUnderlyingNode(),
+                TokenTypes.CLASS_DEF)
+                .findFirstToken(TokenTypes.OBJBLOCK)
+                .findFirstToken(TokenTypes.METHOD_DEF);
+        final DetailAST[] expected = {expectedMethodDefNode.getNextSibling(),
+            expectedMethodDefNode};
+        assertArrayEquals("Result nodes differ from expected", expected, actual);
+    }
+
+    @Test
+    public void testQueryElementNoFollowing() throws Exception {
+        final String xpath = "//CLASS_DEF/following::*";
+        final RootNode rootNode = getRootNode("InputXpathMapperAst.java");
+        final DetailAST[] actual = convertToArray(getXpathItems(xpath, rootNode));
+        assertEquals("Invalid number of nodes", 0, actual.length);
```

Comment:
done

Reply:
Please update all other adder 'assertXxxxx'

Reply:
still not done.

@timurt , please move all unrelated to issue refactoring in separate commit of this PR.
just do:
```
git reset HEAD~1
git add -p
```
keep assertThat in funtional fix only you added to cover issue fix.

Reply:
please check, I tried to move assertThat changes to separate commit

Reply:
good, thanks a lot.

advice for future: such minor commits it is better to put as first commit because chance that it require additional fix is very small, but functional fix might require additional change. Amending non top commit is not trivial in git.

In your case .... if some changes are requested in first commit, you can use https://roman-ivanov.blogspot.com/2019/05/shell-function-to-put-changes-in.html